web-console/src/views/segments-view.tsx

```diff
@@ -133,8 +138,26 @@ export class SegmentsView extends React.Component<SegmentsViewProps, SegmentsVie
     });
   }
 
+  private initTableColumnSelection() {
+    if (localStorage.getItem(segmentTableColumnSelection) == null) {
+      const columns: string[] = ["Segment ID", "Datasource", "Start", "End", "Version", "Partition",
+        "Size", "Num rows", "Replicas", "Is published", "Is realtime", "Is available"];
+      const defaultSetting: any = {};
+      columns.forEach(column => defaultSetting[column] = true);
+      localStorage.setItem(segmentTableColumnSelection, JSON.stringify(defaultSetting));
```

Comment:
Do not write to localStorage unless the user actually hides some columns.

Reply:
Also it would be neater if the object stored in local storage was an array of hidden columns not a mapping of all columns to boolean.